src/ex4/dist/clients/item-client.js

```diff
@@ -0,0 +1,67 @@
+// Create an ItemClient class here. This is what makes requests to your express server (your own custom API!)
+const SERVER_URL = 'http://localhost:8080/todo'
+export default class ItemClient {
+
+    async getTodoList(query = ""){
+        const response = await fetch(`${SERVER_URL}${query}`)
+        const data = await response.json()
+
+        return data
+    }
+
+    async addTodo(value) {
+        await fetch(SERVER_URL, 
+        {
+            method: 'POST',
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({todo: value})
+        })
+    }
+
+    async deleteTodo(index){
+        const response = await fetch(`${SERVER_URL}/${index}`, 
+        {
+            method: 'DELETE'
+        })
+        const data = await response.json()
+        return data.title
+    }
+
+    async getSingleTodo(index){
+        const response = await fetch(`${SERVER_URL}/${index}`)
+        const data = await response.json()
+        
+        return data.todo
+    }
+
+    async editTodoIndex(value,index){
+        const response = await fetch(`${SERVER_URL}/${index}`, 
+        {
+            method: 'PUT',
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({todo: value})
+        })
+        const data = await response.json()
+
+        return data
+    }
+
+    async editCheckTodoIndex(index, status) {
```

Comment:
you probably want to merge editCheckTodoIndex and editTodoIndex into a single updateTodo (by id) endpoint. as you don't want a separate endpoint for each todo property you would want to change

Reply:
ok I see now that both of this method use the same endpoint, however you still want to reuse access to this endpoint, so you could add an additional method for updating a todo, and call it from both editCheckTodoIndex and editTodoIndex